cpp_tests/feature/one_hot_encoder_test.cc

```diff
@@ -39,24 +38,17 @@ class OneHotEncoderTest : public testing::Test {
     mlir_context->appendDialectRegistry(registry);
   }
 
-  static void TearDownTestSuite() {
-    delete host_context;
-    host_context = nullptr;
-    delete mlir_context;
-    mlir_context = nullptr;
-  }
-
-  static tfrt::HostContext *host_context;
-  static MLIRContext *mlir_context;
+  static std::unique_ptr<tfrt::HostContext> host_context;
```

Comment:
Is it intentionally to remove the `TearDownTestSuite(...)` here?

Reply:
If we change `host_context` and `mlir_context` to `unique_ptr`, then we won't need the `TearDownTestSuite(...)` method to explicitly free `host_context` and `mlir_context`. Let's discuss whether we should use C pointers or C++ smart pointer in the next comment.